internal/zinc-persist/src/main/scala-2.12/sbt/internal/inc/consistent/Compat.scala

```diff
@@ -0,0 +1,55 @@
+package sbt.internal.inc.consistent
+
+import java.util.Arrays
+import scala.collection.{ MapLike, SetLike, SortedMap, SortedMapLike }
+import scala.collection.generic.{
+  CanBuildFrom,
+  GenericTraversableTemplate,
+  MapFactory,
+  SeqFactory,
+  SetFactory,
+  SortedMapFactory
+}
+
+// some simple compatibility shims for 2.12 so we don't need to depend on collection-compat
+object Compat {
+  type Factory[-A, +C] = CanBuildFrom[Nothing, A, C]
+
+  implicit def sortedMapFactoryToCBF[CC[A, B] <: SortedMap[A, B] with SortedMapLike[
+    A,
+    B,
+    CC[A, B]
+  ], K: Ordering, V](f: SortedMapFactory[CC]): Factory[(K, V), CC[K, V]] =
+    new f.SortedMapCanBuildFrom
```

Comment:
The auto-formatter is awful. All of this was so much more readable before the auto-formatter ruined it.